packages/vite/src/node/plugins/html.ts

```diff
@@ -338,8 +339,11 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
                 if (!isExcludedUrl(url)) {
                   if (node.tag === 'link' && isCSSRequest(url)) {
                     // CSS references, convert to import
-                    js += `\nimport ${JSON.stringify(url)}`
-                    shouldRemove = true
+                    styleUrls.push({
+                      url,
+                      start: node.loc.start.offset,
+                      end: node.loc.end.offset
+                    })
```

Comment:
I'm still unsure about the order change in the JS that this is going to produce. IIUC, before if the user had first the CSS styles, these were the first imports in the JS, and now they will be imported at the end after all other JS tags. I wonder if we should mark the JS position by adding something like
```
  js += `${STYLE_IMPORT_MARKER_}${styleUrlId}`
```
And then we can replace it by `\nimport ...` or by `` depending on the result of the URL resolution.